pkg/iptableswrapper/mocks/iptables_maps.go

```diff
@@ -41,10 +41,16 @@ func (ipt *MockIptables) Exists(table, chainName string, rulespec ...string) (bo
 }
 
 func (ipt *MockIptables) Insert(table, chain string, pos int, rulespec ...string) error {
+	// We only have -N chainName in the map for chains we manually create
+	if !strings.HasPrefix(chain, "AWS-SNAT-CHAIN") && !strings.HasPrefix(chain, "AWS-CONNMARK-CHAIN") {
+		pos = pos - 1
```

Comment:
This looks strange. It feels like it should be the caller's responsibility to pass the right position and we should not have any logic based on the string name of the chain

Reply:
This is because we can only insert chain creation rule (`-N chain`) into the map for chains we manually create (e.g. `AWS-SNAT-CHAIN`). This function will add all the `-d CIDR` rules to the position right after the chain creation rule. We insert a rule to the `FORWARD` chain in the `filter` table at https://github.com/aws/amazon-vpc-cni-k8s/blob/3be57994609107d9330a15c6a42e42ab1df9f107/pkg/networkutils/network.go#L284. Since this chain is an existing chain rather than one we explicitly create, we do not capture the `-N` rule in the map, hence adding to `pos=1` will result in an out of bounds exception.

Reply:
Update: resolved this